File: server/app/tables/novel.py
import sqlalchemy

# Database table definitions.
metadata = sqlalchemy.MetaData()

novels = sqlalchemy.Table(
    "crawler_novels",
    metadata,
    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
    sqlalchemy.Column("name", sqlalchemy.String),
    sqlalchemy.Column("pic", sqlalchemy.String),
    sqlalchemy.Column("summary", sqlalchemy.Text),
    sqlalchemy.Column("author", sqlalchemy.String),
    sqlalchemy.Column("tag_id", sqlalchemy.Integer),
    sqlalchemy.Column("sum_words", sqlalchemy.Integer),
    sqlalchemy.Column("status", sqlalchemy.Boolean),
    sqlalchemy.Column("created_at", sqlalchemy.DateTime),
    sqlalchemy.Column("updated_at", sqlalchemy.DateTime),
    sqlalchemy.Column("fingerprint", sqlalchemy.String),
    sqlalchemy.Column("is_deleted", sqlalchemy.Boolean),
    sqlalchemy.Column("last_chapter_id", sqlalchemy.Integer)
)

File: crawler/src/models/novel.js
const Sequelize = require('sequelize');
const db = require('./db');

const { STRING, INTEGER, BOOLEAN, TEXT } = Sequelize.DataTypes;

module.exports = db.define('crawler_novel', {
    name: {
        type: STRING,
        required: true
    },
    fingerprint: {
        // name + author md5
        type: STRING,
        unique: true
    },
    // 缩略图
    pic: {
        type: STRING
    },
    summary: {
        type: TEXT,
        required: true
    },
    author: {
        type: STRING,
        required: true
    },
    tag_id: {
        type: INTEGER,
        defaultValue: 0
    },
    sum_words: {
        type: INTEGER,
        defaultValue: 0
    },
    status: {
        type: BOOLEAN,
        defaultValue: false
    },
    is_deleted: {
        type: BOOLEAN,
        defaultValue: false
    },
    last_chapter_id: {
        type: INTEGER
    }
}, {
    timestamps: true
});